Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts

```diff
@@ -0,0 +1,168 @@
+    return packageType === 'runtime' ? 'dncr' : 'dncs';
+}
+
+function getLocalTool(packageType: string, version:string): string {
+    console.log(taskLib.loc("CheckingToolCache"));
+    let cachedToolName = getCachedToolName(packageType);
+    return toolLib.findLocalTool(cachedToolName, version);
+}
+
+async function acquireDotNetCore(packageType: string, version: string): Promise<string> {
+    let downloadUrls = getDownloadUrls(packageType, version);
+    let downloadPath: string;
+
+    try {
+        // try primary url
+        if (!!downloadUrls[0]) {
+            console.log(taskLib.loc("DownloadingPrimaryUrl", downloadUrls[0]));
+            downloadPath = await toolLib.downloadTool(downloadUrls[0]);
+        }
+    } catch (error1) {
+        console.log(taskLib.loc("PrimaryUrlDownloadFailed", error1));
+        try {
+            // try secondary url
+            if (!!downloadUrls[1]) {
+                console.log(taskLib.loc("DownloadingSecondaryUrl", downloadUrls[1]));
+                downloadPath = await toolLib.downloadTool(downloadUrls[1]);
+            }
+        } catch (error2) {
+            console.log(taskLib.loc("LegacyUrlDownloadFailed", error2));
+            throw taskLib.loc("DownloadFailed", packageType, version);
+        }
+    }
+
+    // extract
+    let extPath: string;
+    console.log(taskLib.loc("ExtractingPackage", downloadPath));
+    if (taskLib.osType().match(/^Win/)) {
+        extPath = await toolLib.extractZip(downloadPath);
+    }
+    else {
+        extPath = await toolLib.extractTar(downloadPath);
+    }
+
+    // cache tool
+    let cachedToolName = getCachedToolName(packageType);
+    console.log(taskLib.loc("CachingTool"));
+    let cachedDir =  await toolLib.cacheDir(extPath, cachedToolName, version);
+    console.log(taskLib.loc("SuccessfullyInstalled", packageType, version));
+    return cachedDir;
+}
+
+function getDownloadUrls(packageType: string, version: string): string[] {
+    let scriptRunner: trm.ToolRunner;
+    let primaryUrlSearchString: string;
+    let legacyUrlSearchString: string;
+
+    console.log(taskLib.loc("GettingDownloadUrls", packageType, version));
+    if(taskLib.osType().match(/^Win/)) {
+        let escapedScript = path.join(__dirname, 'externals', 'install-dotnet.ps1').replace(/'/g, "''");
+        let command = `& '${escapedScript}' -Version ${version} -DryRun`
+        if(packageType === 'runtime') {
+            command = command.concat(" -SharedRuntime");
+        }
+
+        let powershellPath = taskLib.which('powershell', true);
+        scriptRunner = taskLib.tool(powershellPath)
+            .line('-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command')
+            .arg(command);
+
+        primaryUrlSearchString = "dotnet-install: Primary - ";
+        legacyUrlSearchString = "dotnet-install: Legacy - ";
+    } else {
+        let escapedScript = path.join(__dirname, 'externals', 'install-dotnet.sh').replace(/'/g, "''");
```

Comment:
saw this pattern in NodeTool task and though no harm in using it - they might have seen some issue earlier.